### PR TITLE
rgblink identifies patches' PC sections after reading all sections

### LINKED ISSUE #794: JR in a LOAD FRAGMENT can fail to link
```
SECTION  "main", ROM0
LOAD FRAGMENT "test", SRAM
ENDL

SECTION "alt", ROM0
LOAD FRAGMENT "test", SRAM
	jr Label
Label:
ENDL
```

This may report `jr target out of reach` or segfault depending on your system.

### PR DESCRIPTION
**Pull request by Rangi42**
Fixes #794

### CODE REVIEW

File: src/link/object.c [hunk 1]
```diff
@@ -305,14 +303,24 @@ static void readPatch(FILE *file, struct Patch *patch, char const *fileName, cha
 		     feof(file) ? "Unexpected end of file" : strerror(errno));
 }
 
+/**
+ * Sets a patch's pcSection from its pcSectionID.
+ * @param patch The struct to fix
+ */
+static void fixPatchSection(struct Patch *patch, struct Section *fileSections[])
```
Comment (ISSOtm) [inline src/link/object.c:310]: "Fix" implies that it was broken before, so I'd prefer a name that better conveys that it's actually filling that data in, and for the first time.

**Comment by Rangi42** [comment]
@ISSOtm Please take a look at this; I don't know why *some* of the CI tests fail to build pokered.

**Review by ISSOtm — CHANGES_REQUESTED** [review]

File: src/link/object.c (hunk 1 above)
Comment (Rangi42) [reply src/link/object.c:310]: `linkPatchToPCSect`?

Comment (ISSOtm) [reply src/link/object.c]: Sounds good, yep

File: src/link/object.c [hunk 2]
```diff
@@ -279,8 +279,7 @@ static void readPatch(FILE *file, struct Patch *patch, char const *fileName, cha
 	tryReadlong(patch->pcSectionID, file,
 		    "%s: Unable to read \"%s\"'s patch #%" PRIu32 "'s PC offset: %s",
 		    fileName, sectName, i);
-	patch->pcSection = patch->pcSectionID == (uint32_t)-1 ? NULL
-							      : fileSections[patch->pcSectionID];
+	patch->pcSection = NULL; /* set later by linkPatchToPCSect */
```
Comment (ISSOtm) [inline src/link/object.c:282]: If it's going to be filled in, why is this initialization needed? If it isn't, the comment should be enough.

**Review by ISSOtm — CHANGES_REQUESTED** [review]

**Review by ISSOtm — APPROVED** [review]

**Pull request merged**